dimod/sampleset.py

```diff
@@ -997,6 +1002,90 @@ def aggregate(self):
+                case the given order will be maintained.
+
+        Returns:
+            :obj:`.SampleSet`: A new sample set with the samples appended. If
+            `samples_like` is another sample set then the returned sample set
+            has the sum of the energies.
+
+        Examples:
+
+            >>> sampleset = dimod.SampleSet.from_samples([{'a': -1, 'b': +1},
+            ...                                           {'a': +1, 'b': +1}],
+            ...                                          dimod.SPIN,
+            ...                                          energy=[-1.0, 1.0])
+            >>> new = sampleset.append_variables({'c': -1})
+            >>> print(new)
+               a  b  c energy num_oc.
+            0 -1 +1 -1   -1.0       1
+            1 +1 +1 -1    1.0       1
+            ['SPIN', 2 rows, 2 samples, 3 variables]
+
+            Append another sampleset to the original above.
+
+            >>> another = dimod.SampleSet.from_samples([{'c': -1, 'd': +1},
+            ...                                         {'c': +1, 'd': +1}],
+            ...                                        dimod.SPIN,
+            ...                                        energy=[-2.0, 1.0])
+            >>> new = sampleset.append_variables(another)
+            >>> print(new)
+               a  b  c  d energy num_oc.
+            0 -1 +1 -1 +1   -3.0       1
+            1 +1 +1 +1 +1    2.0       1
+            ['SPIN', 2 rows, 2 samples, 4 variables]
+
+        """
+        samples, labels = as_samples(samples_like)
+
+        num_samples = len(self)
+
+        # we don't handle multiple values
+        if samples.shape[0] == num_samples:
+            # we don't need to do anything, it's already the correct shape
+            pass
+        elif samples.shape[0] == 1 and num_samples:
+            samples = np.repeat(samples, num_samples, axis=0)
+        else:
+            msg = ("mismatched shape. The samples to append should either be "
+                   "a single sample or should match the length of the sample "
+                   "set. Empty sample sets cannot be appended to.")
+            raise ValueError(msg)
+
+        # append requires the new variables to be unique
+        variables = self.variables
+        if any(v in variables for v in labels):
+            msg = "Appended samples cannot contain variables in sample set"
+            raise ValueError(msg)
+
+        new_variables = list(variables) + labels
+        new_samples = np.hstack((self.record.sample, samples))
+
+        new = type(self).from_samples((new_samples, new_variables),
+                                      self.vartype,
+                                      info=dict(self.info),  # make a copy
+                                      sort_labels=sort_labels,
+                                      **self.data_vectors)
+
+        if isinstance(samples_like, SampleSet):
+            new.record.energy += samples_like.record.energy
```

Comment:
It's weird to special-handle energies, out of all info vectors. What about `num_occurrences`, for example?

Reply:
Adding `num_occurences` does not make sense, energies is meaningful at least. Other option is to just ignore them all

Reply:
I don't agree with ignoring all.
energy is the only thing that changes with fixedvariables. num_occurrences should stay the same

Reply:
Energy does not change in fixed variables because we use the offset

Reply:
Actually, ignoring all is the only option that makes sense in the `append_variables` method.

Reply:
ok I get it. samples on their own don't exist anyway. yes ignore all ftw